compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -933,6 +933,12 @@ impl<'tcx> PolyExistentialTraitRef<'tcx> {
     }
 }
 
+impl rustc_errors::IntoDiagnosticArg for PolyExistentialTraitRef<'_> {
+    fn into_diagnostic_arg(self) -> rustc_errors::DiagnosticArgValue<'static> {
+        self.to_string().into_diagnostic_arg()
```

Comment:
Isn't there a derive for this?

Reply:
Er, a macro, I mean. Something that makes a diagnostic arg impl via string.

Reply:
I see https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_errors/diagnostic_impls.rs.html#57-80. Would it make sense to have a blanket impl for `T: Display` 🤔

Reply:
I guess it would make sense to export that macro or even do a blanket impl but that can be a follow up PR since there are probably many other places that would need to be adjusted

Reply:
+1 on the follow-up 😃